src/edgeProxy/public/javascripts/uio.js

```diff
@@ -1,19 +1,166 @@
+/* global getCookieValue */
+
 "use strict";
 
-fluid.uiOptions(".flc-prefsEditor-separatedPanel", {
-    auxiliarySchema: {
-        terms: {
-            "templatePrefix": "lib/infusion/src/framework/preferences/html",
-            "messagePrefix": "lib/infusion/src/framework/preferences/messages"
-        },
-        "fluid.prefs.tableOfContents": {
-            enactor: {
-                "tocTemplate": "lib/infusion/src/components/tableOfContents/html/TableOfContents.html",
-                "tocMessage": "lib/infusion/src/framework/preferences/messages/tableOfContents-enactor.json",
-                ignoreForToC: {
-                    "overviewPanel": ".flc-overviewPanel"
-                }
+fluid.registerNamespace("fluid.prefs.edgeProxy");
+
+fluid.defaults("fluid.dataSource.noEncoding", {
+    gradeNames: ["fluid.dataSource", "fluid.dataSource.writable"],
+    components: {
+        encoding: {
+            type: "fluid.dataSource.encoding.none"
+        }
+    }
+});
+
+// Edge Proxy Store
+fluid.defaults("fluid.prefs.edgeProxyStore", {
+    gradeNames: ["fluid.dataSource.noEncoding", "fluid.modelComponent"],
+    members: {
+        isFreshLogon: "@expand:fluid.prefs.edgeProxyStore.getFreshLogon()"
+    },
+    model: {
+        isLoggedIn: false
+    },
+    components: {
+        unauthedStore: {
+            type: "fluid.prefs.cookieStore",
+            options: {
+                writable: true
             }
         }
+    },
+    dynamicComponents: {
+        authedStore: {
+            source: "{that}.model.isLoggedIn",
+            type: "fluid.prefs.pdsStore"
+        }
+    },
+    listeners: {
+        "onCreate.updateLoggedInState": {
+            listener: "fluid.prefs.edgeProxyStore.updateLoggedInState",
+            args: ["{that}"]
+        },
+        "onRead.impl": {
+            listener: "fluid.prefs.edgeProxyStore.get",
+            args: ["{that}"]
+        },
+        "onWrite.impl": {
+            listener: "fluid.prefs.edgeProxyStore.set",
+            args: ["{that}", "{arguments}.0"]   // settings
+        }
+    },
+    modelListeners: {
+        isLoggedIn: {
+            listener: "fluid.prefs.edgeProxyStore.updateUnauthedSettings",
+            args: ["{that}", "{prefsEditorLoader}", "{change}.value"],
+            excludeSource: "init"
+        }
     }
 });
+
+// Update the fresh logon state at the page load. Check if PDS_freshLogon is a query parameter of the page url.
+// If yes, return true. Otherwise, return false.
+fluid.prefs.edgeProxyStore.getFreshLogon = function () {
+    const url = new URL(window.location);
+    const isFreshLogon = url.searchParams.get("PDS_freshLogon") === "true";
+
+    if (isFreshLogon) {
+        // Remove "PDS_freshLogon" from query strings not to confuse users
+        url.searchParams.delete("PDS_freshLogon");
+        window.history.replaceState({}, "", url.href);
```

Comment:
Does this affect the back button?

Reply:
It doesn't affect the back button as it modifies the current history entry. In our case that this modification is hit when being redirected back from Google sign in. The back button will lead back to the Google sign in page.